src/lib/modules/SearchControls/overridden/SearchSortOrderElementMobile.js

```diff
@@ -0,0 +1,55 @@
+import { Button, Icon } from 'semantic-ui-react';
+import PropTypes from 'prop-types';
+import React, { Component } from 'react';
+
+export class SearchSortOrderElementMobile extends Component {
+  constructor(props) {
+    super(props);
+    this.buttons = {};
+  }
+
+  sortChangeHandler = (onValueChange, currentSortOrder, options) => {
+    const value =
+      currentSortOrder === options[0].value
+        ? options[1].value
+        : options[0].value;
+    onValueChange(value);
+  };
+
+  render() {
+    const { currentSortOrder, options, onValueChange } = this.props;
```

Comment:
Aren't the options fixed after received? If yes, we could avoid all this recomputation on every render and fill in the `this.buttons` property on the constructor. Then, you leave here only the `currentSortOrder` prop to control when you will need to rerender :)

Reply:
done!